### DISCUSSION TITLE
Is a oqsprovider integration for openSSH possible w/o recompilation?

### DISCUSSION THREAD

**Discussion by adakac** [open-quantum-safe/oqs-provider#303] [Q&A] [answered]
Hey there,

I am currently playing around with your project and haven't gotten through this yet. I know there would be the possibility to edit and compile my own openssh version (I know that there is a [repo](https://github.com/open-quantum-safe/openssh) for that). However, do you guys have an easier option (f.e. using `KexAlgorithms` in the `ssh_config`) on how to get openssh to recognize f.e. kyber-512-sha256 as a valid KEX?

Big thanks in advance for any pointers! :)

**Comment by baentsch** [comment, marked as answer]
Hmm -- `oqsprovider` is a plugin for `openssl`. `openssl` in turn IMO is no longer the default crypto provider for `openssh`, AFAIK for sure not for the KEX logic in `openssh`. Hence, I'm afraid the answer to your question is "No". 

If you'd find a way (or any of my --very basic-- understanding of `openssh` is wrong), I'd be really curious to hear about it: We're not really having a maintainer for `oqs-openssh`, so being able to handle this via `oqsprovider` would be great.

**Reply by adakac** [reply]
Hey there, thanks for elaborating! Sadly I couldn't find another way in my limited time playing around with it.. I used the self-recompile way of the oqs-ssh project at last.
